### ISSUE TITLE
KEEP LOOSING DATA, Fix this Broken App...

### ISSUE THREAD

**Issue by jjiglesiasg** [facebook/Rapid#1153]
### Description

Rapid crash quite often and once you restart it could NOT upload the "local saved" data, and your forcefully needs to mark the X and Loose whatever you have done and Restart all over again.

Today I have lost 3 times the edicted data (More than 300 items per Changeset, Buildings, Areas, Roads, etc, etc. Even using the Power user feature. I mean more than 700 items lost.

The worst disrispect for any mapper is a program that force you to loose YOUR TIME and Data. Something is WRONG with this app. Because than DO NOT happen with ID.

Really is ANOYING to work hours just to trash your work. We mappers are NOT JOKING here...

### Version

2.1.1

### Steps to reproduce

FIX THE APP

### The browser URL at the time you encountered the bug

https://rapideditor.org/rapid#background=Bing&datasets=fbRoads,msBuildings&disable_features=boundaries&id=w705008121&map=17.32/-17.77286/-63.22823&poweruser=true

**Comment by bhousel** [comment]
Sorry to hear you are still having problems with Rapid crashing.  I really do want to fix the app, but I'm not sure where to start. Rapid is very stable for me on my 3 years old Macbook with Google Chrome, and I don't think I've ever lost data.

If you are able to get Rapid to crash frequently, the best thing that can help us out is if you open the browser developer tools and send us any error messages that pop up there, or a screenshot.   This was my request in https://github.com/facebook/Rapid/issues/1066#issuecomment-1699285090 .

I have been working on the internals of the edit system a lot over the past few weeks.. I'm hoping that work will get us closer to being able to store the really large backups in indexdb instead of localstorage. (this is #1055, #187, #1078)

Also remember to not open Rapid in multiple browser tabs - only one of them will be able to write backups.

**Comment by jjiglesiasg** [comment]
Again, More data losy. That is Totally Annoying

![Screenshot (110)](https://github.com/facebook/Rapid/assets/22849578/0d6b3fc8-b058-4b91-8ee5-2c3f5d52db23)

**Comment by bhousel** [comment]
Yes, in the screenshot it shows the warning that your data isn't being backed up..

I don't know why this is. (note for me: the [guidance here](https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria) is the best summary I could find about it.) 
My goal is to move the backups to a better place.

<img width="573" alt="Screenshot 2023-10-16 at 3 38 25 PM" src="https://github.com/facebook/Rapid/assets/38784/7c88afd2-24d5-46cd-a591-c57dcfa09757">

**Comment by bhousel** [comment]
I think these are hopefully fixed by #1658 - it would explain why you keep seeing freezes that I'm not able to repro.
